File: Objective-C/EventCapture/EventCapture/ViewController.m
//
//  ViewController.m
//  EventCapture
//
//  Created by zidon on 16/8/17.
//  Copyright © 2016年 zidon. All rights reserved.
//

#import "ViewController.h"
#import "RunloopEventCapture.h"

@interface ViewController ()<RunloopEventCaptureDelegate>
{
    RunloopEventCapture *_runloop;
}
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.］
    
    UIButton *btn=[UIButton buttonWithType:UIButtonTypeCustom];
    btn.frame=CGRectMake(10, 20, 100, 30);
    btn.backgroundColor=[UIColor redColor];
    [btn setTitle:@"source" forState:UIControlStateNormal];
    [btn addTarget:self action:@selector(jojo1) forControlEvents:UIControlEventTouchUpInside];
    [self.view addSubview:btn];
    
    _runloop=[RunloopEventCapture sharedRunLoopWorkDistribution];
    _runloop.delegate=self;
    
    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(10 * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
        [self jojo1];
    });
    
}

- (void)jojo1 {
    
    [_runloop addTask:^BOOL{
        return YES;
    } withKey:@"lala"];
    
    
}

//这个过程是异步的，注意操作UI的时候要回到主线程
-(void)testUploadCaptureUserData:(NSString *)key
{
    NSLog(@":::%@",key);
}

-(void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event
{
    
    NSLog(@"lalala");
}


@end

File: Swift/贝塞尔曲线/贝塞尔曲线/ViewController.swift
//
//  ViewController.swift
//  贝塞尔曲线
//
//  Created by zidon on 16/4/6.
//  Copyright © 2016年 zidon. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    var berthV:berthView?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        berthV=berthView.init(frame:CGRectMake(100,100,250,250))
        berthV?.backgroundColor=UIColor.brownColor()
        self.view.addSubview(berthV!)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

}

